base/data/Campaign.js

```diff
@@ -169,52 +169,35 @@ Campaign.hideAdverts = (topCampaign, campaigns) => {
  * Get all ads associated with the given campaigns
  * @param {Campaign} topCampaign 
  * @param {?Campaign[]} campaigns 
- * @returns {Advert[]}
+ * @param {?KStatus} status
+ * @returns PromiseValue(Advert[])
  */
-Campaign.fetchAds = (topCampaign, campaigns) => {
-    
-    let {
-        status,
-        'gl.status': glStatus
-    } = DataStore.getValue(['location', 'params']) || {};
-    if ( ! status) status = (glStatus || C.KStatus.PUB_OR_ARC);
+Campaign.fetchAds = (topCampaign, campaigns, status=KStatus.DRAFT) => {
 
-    let ads = [];
-    // Get ads for top campaign
-    let q = SearchQuery.setProp(new SearchQuery(), "campaign", topCampaign.id).query;
-    let pvAds = ActionMan.list({type: C.TYPES.Advert, status, q});
-    if (pvAds.value) {
-        List.hits(pvAds.value).forEach(ad => {
-            ads.push(ad);
-        });
+    let sq = SearchQuery.setProp(new SearchQuery(), "campaign", topCampaign.id);
+    if (campaigns) {
+        let sq2 = SearchQuery.setPropOr(new SearchQuery(), "campaign", campaigns.map(c => c && c.id).filter(x => x));
+        sq = SearchQuery.or(sq, sq2);
     }
-    // Get ads for associated campaigns
-    campaigns && campaigns.forEach(campaign => {
-        if (!campaign || !campaign.id) return;
-        let qOtherAds = SearchQuery.setProp(new SearchQuery(), "campaign", campaign.id).query;
-        let pvOtherAds = ActionMan.list({type: C.TYPES.Advert, status, q:qOtherAds});
-        if (pvOtherAds.value) {
-            List.hits(pvOtherAds.value).forEach(ad => {
-                if (!ads.includes(ad)) ads.push(ad);
-            });
-        }
-    });
+    const q = sq.query;
+    const pvAds = ActionMan.list({type: C.TYPES.Advert, status, q});
 
-    return ads;
+    return pvAds;
 }
 
 /**
  * Get a list of adverts that the impact hub should show for this campaign
  * @param {Campaign} topCampaign the subject campaign
  * @param {?Campaign[]} campaigns any other campaigns with data to use (for advertisers or agencies)
  * @param {?Advert[]} presetAds use a preset list of ads instead of fetching ourselves
- *  * @param {?Boolean} showNonServed show ads that have never served - overrides GET parameter of same name if true
+ * @param {?Boolean} showNonServed show ads that have never served - overrides GET parameter of same name if true
  * @param {?Boolean} nosample disable automatic sampling - overrides GET parameter of same name if true
  * @returns {Advert[]} adverts to show
  */
-Campaign.advertsToShow = (topCampaign, campaigns, presetAds, showNonServed, nosample) => {
+Campaign.advertsToShow = (topCampaign, campaigns, status=KStatus.DRAFT, presetAds, showNonServed, nosample) => {
```

Comment:
Please switch to named parameters. It's our usual preference, unless there's only one or two parameters. Thanks.